docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,200 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+Async programming is traditionally associated with callbacks and a general lack of straightforward code.  One might even say that straightforward code and scalable code are often at odds with one another.  This is not a problem with the C#/VB async model!  It is a language-level construct which hides complexity and allows you to focus on the problem at hand without making the code confusing in the process.
```

Comment:
> It seems odd to refer it the model as the "C#/VB async model". Task is not a language thing. The syntax is.

I'd argue it's far more than a syntax thing.Even though C#/VB and F# implement the same overall model (Promises), they approach it differently.  The F# approach is based on computation expressions which treat async operations as expressions of work to be done in some asynchronous context.

The C#/VB model much more straightforward conceptually, using objects which represent the operation running right now (`Task` and `Task<T>`) as things you pass around and eventually wait to complete.  F#'s async does not use the same types either; it uses the `Async<'a> type, which although is used roughly the same way does result in things being a little different (namely it makes the compiler even stricter than it is without async workflows).

The least subtle difference in terms of use is that you cannot nest tasks (e.g. `await Foo(await GetResult(), bar);` is impossible in F#).  There are also numerous more subtle (and devious) things you can do with C#/VB which can cause the code to not run asynchronously.  It's literally impossible to have that happen in F# unless you input that async operation into the `Async.RunSynchronously` function.

I'd like to avoid conflating anything F#-related in this document to avoid giving the impression that they're operating similarly.